src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,47 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  static removeDead() {
+    const aliveAnimals = this.alive.filter((animal) => animal.health > 0);
+
+    this.alive = [...aliveAnimals];
+  }
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(health, name) {
+    super(health, name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(health, name) {
+    super(health, name);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (!animal.hidden && animal.constructor.name === 'Herbivore') {
+      animal.health -= 50;
+    }
```

Comment:
The condition in the if statement should only check if the animal is an instance of Herbivore and not hidden. There's no need to compare the constructor name as a string; instead, use `instanceof` for a more robust check.